### PR TITLE
Document Provider Specific Configuration, Small Readme Update

### LINKED ISSUE #26: Documentation: Update Readme to reflect admin page + document usage with a few providers (keycloak, authelia)
I think we should provide a few example docs that demonstrate usage with as many providers as have been tested successfully.

Presently, at least the following providers have been tested succesfully:

- Keycloak:
  - successfully configured with the full suite of role-based access permissions by 9p4
  - this is demo'd in the readme. 
- Authelia:
  - successfully configured basic authentication (no role checking)
  - pending #24 , will also support role checking: https://github.com/9p4/jellyfin-plugin-sso/issues/23#issuecomment-1112237616 
  - The documentation of this should be formalized somewhere

Besides these two, which are probably most important as they constitute two of the most the most popular self-host Identity Providers it might be helpful to document additional providers, eg

- Authentik? (New-ish self-host ID provider) (Let an authentik user figure this out)
- Google ?
  - successfully configured basic auth, with some limitations according to 9p4

### PR DESCRIPTION
**Pull request by matthewstrasiotto**
Partially resolves #26

Still needs keycloak documentation

### CODE REVIEW

**Comment by 9p4** [comment]
I've added Keycloak.

File: providers.md [hunk 1]
```diff
@@ -57,14 +57,13 @@ identity_providers:
...
-On the end of `jellyfin`, we need to configure an authelia provider as follows:
+On Jellyfin's end, we need to configure an Authelia provider as follows:
 
-In order to test group membership, we need to request authelia's OIDC scope `groups`,
-which we will use to check user roles.
+In order to test group membership, we need to request Authelia's `groups` OIDC scope, which we will use to check user roles.
 
 ```yaml
 authelia:
-  OidEndpoint: https://authelia.example.com/.well-known/openid-configuration/
```
Comment (matthewstrasiotto) [inline providers.md]: This is actually the endpoint you need to specify

Comment (matthewstrasiotto) [reply providers.md]: oh, idk maybe it's not necessary

**Comment by 9p4** [comment]
Saw this on HN... is it important enough to add docs about? https://github.com/zitadel/zitadel

**Comment by matthewstrasiotto** [comment]
> Saw this on HN... is it important enough to add docs about? https://github.com/zitadel/zitadel

looks neat from a casual glance at the repo, but no need to rush it into this PR, we can always add providers later, or let users document them.

File: providers.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+- Edit / Update your Jellyfin OAuth provider
+- Under **"Advanced Protocol Settings"**, add the **Group Membership** Scope
+
+  ![image](img/authentik-config-05.jpg)
+
+### Jellyfin's Config
+
+On Jellyfin's end, we need to configure an Authentik provider as follows:
+
+In order to test group membership, we need to request authelia's OIDC scope `groups`, which we will use to check user roles.
```
Comment (NeroPcStation) [inline providers.md:124]: should be Authentik

**Comment by 9p4** [comment]
Seems like Authentik is meant to be stylized as all lowercase: `authentik` not `Authentik`

**Comment by matthewstrasiotto** [comment]
can this be merged after a rebase onto main?

**Comment by strazto** [comment]
Believe it or not I was at work trying to test our SAML config works against keycloak and document the setup and I wound up referring to this plugin's keycloak writeup to find a working configuration, thanks again @9p4 🤣

**Pull request merged**